server/app/views/admin/programs/ProgramTranslationView.java

```diff
@@ -289,12 +238,14 @@ private ImmutableList<DomContent> getApplicantVisibleProgramDetailFields(
                         .getInputTag(),
                     program.localizedName()));
 
-    // Always show the long description field if north star is off
-    // If north star is enabled, only show the long description field if the program is not a common
-    // intake program
-    boolean northStarIsOff = !settingsManifest.getNorthStarApplicantUi(request);
-    boolean isNotCommonIntake = !program.isCommonIntakeForm();
-    if (northStarIsOff || isNotCommonIntake) {
+    // On north star, only default programs have a long description. Whereas when north star is off,
+    // both default programs and common intake forms have long description.
+    boolean northStarEnabled = settingsManifest.getNorthStarApplicantUi(request);
+    ProgramType programType = program.programType();
+    if ((northStarEnabled && programType.equals(ProgramType.DEFAULT))
+        || (!northStarEnabled
+            && (programType.equals(ProgramType.DEFAULT)
```

Comment:
suggestion (blocking): This `if` statement is hard to read, which isn't surprising because this logic in convoluted! I have two suggestions:

1. assign these two clauses to vars and then use them in the if statement
2. check that the program is not an external program, instead of checking that it is default or common intake

so something like:
```
boolean northStarOnDefaultProgram = northStarEnabled && programType.equals(ProgramType.DEFAULT);
boolean northStarOffNonExternalProgram =  !northStarEnabled && !programType.equals(ProgramType.EXTERNAL);

if (northStarOnDefaultProgram || northStarOffNonExternalProgram) {
...
}

Reply:
I think (1) is more correct. Problem with having 'not this' means if a new type is added then the functionality is broken. It's a better practice to check for the exact condition